contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.2.0--2.3.0.sql

```diff
@@ -203,785 +197,200 @@ SELECT
-		FROM pg_catalog.pg_auth_members AS Authmbr
-		INNER JOIN pg_catalog.pg_roles AS Base1 ON Base1.oid = Authmbr.roleid
-		INNER JOIN pg_catalog.pg_roles AS Base2 ON Base2.oid = Authmbr.member
-		INNER JOIN sys.babelfish_authid_login_ext AS Ext1 ON Base1.rolname = Ext1.rolname
-		INNER JOIN sys.babelfish_authid_login_ext AS Ext2 ON Base2.rolname = Ext2.rolname
-		WHERE Ext1.type = 'R'
-		ORDER BY ServerRole, MemberName;
-	END
-	-- If a valid server role is specified, return its member info
-	-- If the role is a SQL server predefined role (i.e. serveradmin), 
-	-- do not raise an error even if it does not exist
-	ELSE IF EXISTS (SELECT 1
-					FROM sys.babelfish_authid_login_ext
-					WHERE (rolname = RTRIM(@srvrolename)
-					OR lower(rolname) = lower(RTRIM(@srvrolename)))
-					AND type = 'R')
-					OR lower(RTRIM(@srvrolename)) IN (
-					'serveradmin', 'setupadmin', 'securityadmin', 'processadmin',
-					'dbcreator', 'diskadmin', 'bulkadmin')
-	BEGIN
-		SELECT CAST(Ext1.rolname AS sys.SYSNAME) AS 'ServerRole',
-			   CAST(Ext2.rolname AS sys.SYSNAME) AS 'MemberName',
-			   CAST(CAST(Base2.oid AS INT) AS sys.VARBINARY(85)) AS 'MemberSID'
-		FROM pg_catalog.pg_auth_members AS Authmbr
-		INNER JOIN pg_catalog.pg_roles AS Base1 ON Base1.oid = Authmbr.roleid
-		INNER JOIN pg_catalog.pg_roles AS Base2 ON Base2.oid = Authmbr.member
-		INNER JOIN sys.babelfish_authid_login_ext AS Ext1 ON Base1.rolname = Ext1.rolname
-		INNER JOIN sys.babelfish_authid_login_ext AS Ext2 ON Base2.rolname = Ext2.rolname
-		WHERE Ext1.type = 'R'
-		AND (Ext1.rolname = RTRIM(@srvrolename) OR lower(Ext1.rolname) = lower(RTRIM(@srvrolename)))
-		ORDER BY ServerRole, MemberName;
-	END
-	-- If the specified server role is not valid
-	ELSE
-		RAISERROR('%s is not a known fixed role.', 16, 1, @srvrolename);
-END;
+    query1 := format('alter extension babelfishpg_tsql drop function %s.%s', schema_name, func_name);
+    query2 := format('drop function %s.%s', schema_name, func_name);
+    execute query1;
+    execute query2;
+EXCEPTION
+    when object_not_in_prerequisite_state then --if 'alter extension' statement fails
+        GET STACKED DIAGNOSTICS error_msg = MESSAGE_TEXT;
+        raise warning '%', error_msg;
+    when dependent_objects_still_exist then --if 'drop function' statement fails
+        GET STACKED DIAGNOSTICS error_msg = MESSAGE_TEXT;
+        raise warning '%', error_msg;
+end
 $$
-LANGUAGE 'pltsql';
-GRANT EXECUTE ON PROCEDURE sys.sp_helpsrvrolemember TO PUBLIC;
-
--- Need to add parameter for tsql_type_max_length_helper 
-ALTER FUNCTION sys.tsql_type_max_length_helper RENAME TO tsql_type_max_length_helper_deprecated_in_2_3_0;
+LANGUAGE plpgsql;
 
-CREATE OR REPLACE FUNCTION sys.tsql_type_max_length_helper(IN type TEXT, IN typelen INT, IN typemod INT, IN for_sys_types boolean DEFAULT false, IN used_typmod_array boolean DEFAULT false)
-RETURNS SMALLINT
-AS $$
+CREATE OR REPLACE FUNCTION sys.babelfish_conv_date_to_string(IN p_datatype TEXT,
```

Comment:
It would be really hard to review the upgrade script changes by 1-2 reviewers. If you have a spreadsheet containing a list of objects, then we may ask many others to review a reasonable number of objects.

Reply:
may be original writers review their procedures/functions

Reply:
babelfish_conv_date_to_string is already present in upgrade script updated by you. So you can skip it.